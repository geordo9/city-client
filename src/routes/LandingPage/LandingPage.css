.LandingPage {
    margin: 0 auto;
    max-width: 650px;
}

.LandingPage__footer {
    text-align: center;
    
}

.LandingPage section {
    background-color: #c0c0c0;
    border-bottom: 1px solid #39414C;
    padding-top:20px;
    padding-bottom: 20px;
    padding-left: 5px;
    padding-right: 5px;
}

.LandingPage a {
    text-decoration: none;
    color: goldenrod;
    background-color: #202a25;
    border-radius: 10px;
    padding: 5px 15px;
}

.LandingPage img {
    width: 100%
}