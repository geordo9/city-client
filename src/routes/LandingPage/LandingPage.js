import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import './LandingPage.css'


export default class ThingListPage extends Component {
  //would like the link here to connect to the user page but can't figure out how to find user id
  renderUserLink() {
    return (
      <div className='Footer__logged-in'>
        <Link to={`/user`}>
          Go to Your User Page!
        </Link>
      </div>
    )
  }

  renderRegisterLink() {
    return (
      <div>
        <header>
          <h3>Ready to see how your city stacks up? Sign-up!</h3>
        </header>
          <Link
            className='registration-button'
            to='/register'>
            Register
          </Link>
      </div>
    )
  }

  render() {
      return (
        <main role="main" className="LandingPage">
      <header role="banner">
        <h1>City Showdown</h1>
        <h2>A chance to finally stack up your teams against your friends</h2>
      </header>
      <section>
        <header>
            <h3>Stack up the Cities</h3>
        </header>
        <img src={require('./showdown_page.PNG')} alt={"an example of the showdown page"} />
        <p>City Showdown lets you compare your city's playoffs record against another across the four major sports.</p>
      </section>
      <section>
        <header>
            <h3>Yankees or Mets? Cubs or White Sox? Customize your favorite teams!</h3>
        </header>
        <img src={require('./edit_user_page.PNG')} alt={"an example of the edit user page"} />
        <p>Many cities have multiple teams for a single sport. Who are we to overlap the 2005 White Sox title with the Cubbies' win 2016?</p>
      </section>
      <footer className="LandingPage__footer">
        {TokenService.hasAuthToken()
            ? this.renderUserLink()
            : this.renderRegisterLink()}
      </footer>
    </main>
      );
  }
}