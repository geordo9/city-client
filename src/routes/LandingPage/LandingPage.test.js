import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LandingPage from './LandingPage'


describe(`LandingPage component`, () => {
    
    it('renders without crashing', () => {

  const div = document.createElement('div');

  ReactDOM
    .render(
        <MemoryRouter>
            <LandingPage />
        </MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
})

  it('renders a LandingPage by default', () => {
    const wrapper = shallow(<LandingPage /> )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})