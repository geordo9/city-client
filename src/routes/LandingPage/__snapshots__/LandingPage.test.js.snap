// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders a LandingPage by default 1`] = `
<main
  className="LandingPage"
  role="main"
>
  <header
    role="banner"
  >
    <h1>
      City Showdown
    </h1>
    <h2>
      A chance to finally stack up your teams against your friends
    </h2>
  </header>
  <section>
    <header>
      <h3>
        Stack up the Cities
      </h3>
    </header>
    <img
      alt="an example of the showdown page"
      src="showdown_page.PNG"
    />
    <p>
      City Showdown lets you compare your city's playoffs record against another across the four major sports.
    </p>
  </section>
  <section>
    <header>
      <h3>
        Yankees or Mets? Cubs or White Sox? Customize your favorite teams!
      </h3>
    </header>
    <img
      alt="an example of the edit user page"
      src="edit_user_page.PNG"
    />
    <p>
      Many cities have multiple teams for a single sport. Who are we to overlap the 2005 White Sox title with the Cubbies' win 2016?
    </p>
  </section>
  <footer
    className="LandingPage__footer"
  >
    <div>
      <header>
        <h3>
          Ready to see how your city stacks up? Sign-up!
        </h3>
      </header>
      <Link
        className="registration-button"
        replace={false}
        to="/register"
      >
        Register
      </Link>
    </div>
  </footer>
</main>
`;

exports[`LandingPage component renders a ShowdownPage by default 1`] = `
<main
  className="LandingPage"
  role="main"
>
  <header
    role="banner"
  >
    <h1>
      City Showdown
    </h1>
    <h2>
      A chance to finally stack up your teams against your friends
    </h2>
  </header>
  <section>
    <header>
      <h3>
        Stack up the Cities
      </h3>
    </header>
    <img
      alt="an example of the showdown page"
      src="showdown_page.PNG"
    />
    <p>
      City Showdown lets you compare your city's playoffs record against another across the four major sports.
    </p>
  </section>
  <section>
    <header>
      <h3>
        Yankees or Mets? Cubs or White Sox? Customize your favorite teams!
      </h3>
    </header>
    <img
      alt="an example of the edit user page"
      src="edit_user_page.PNG"
    />
    <p>
      Many cities have multiple teams for a single sport. Who are we to overlap the 2005 White Sox title with the Cubbies' win 2016?
    </p>
  </section>
  <footer
    className="LandingPage__footer"
  >
    <div>
      <header>
        <h3>
          Ready to see how your city stacks up? Sign-up!
        </h3>
      </header>
      <Link
        className="registration-button"
        replace={false}
        to="/register"
      >
        Register
      </Link>
    </div>
  </footer>
</main>
`;
